customers/views.py

```diff
@@ -53,30 +45,11 @@ def retrieve(customer_id):
         if customer_id is None:
             customer_list = []
             for selected_customer in Customer.query.all():
-                customer_list.append(
-                    {
-                        "id": f"{selected_customer.id}",
-                        "username": f"{selected_customer.username}",
-                        "first_name": f"{selected_customer.first_name}",
-                        "last_name": f"{selected_customer.last_name}",
-                        "email": f"{selected_customer.email}",
-                        "dob": f"{selected_customer.dob}",
-                        "date_created": f"{selected_customer.date_created}",
-                    }
-                )
+                customer_list.append(selected_customer.to_dict())
             return json.dumps(customer_list), 200
         else:
             selected_customer = Customer.query.filter_by(id=customer_id).first()
-            returning_data = {
-                "id": f"{selected_customer.id}",
-                "username": f"{selected_customer.username}",
-                "first_name": f"{selected_customer.first_name}",
-                "last_name": f"{selected_customer.last_name}",
-                "email": f"{selected_customer.email}",
-                "dob": f"{selected_customer.dob}",
-                "date_created": f"{selected_customer.date_created}",
-            }
-            return json.dumps(returning_data), 200
+            return json.dumps(selected_customer.to_dict()), 200
```

Comment:
don't mix serialization methods, above you used `customerSchema.jsonify(newCustomer)` and here you used `json.dumps(selected_customer.to_dict())`

pick a method and stay consistent